### ISSUE TITLE
AttributeError: 'tqdm' object has no attribute 'pos'

### ISSUE THREAD

**Issue by zyrikby** [tqdm/tqdm#563]
I saw the related bugs and know that they are caused by concurrency. I manually added necessary lock to my child threads but I still get this error from time to time. I know that it is difficult to reproduce however may be a stacktrace will give you some hints.

```bash
File "/home/yury/.pyenv/versions/3.6.5/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
File "az_downloader_threading.py", line 114, in run
    res = self._download_file(queue_item)
File "az_downloader_threading.py", line 134, in _download_file
    time.sleep(0.1)
File "/home/yury/.pyenv/versions/3.6.5/lib/python3.6/site-packages/tqdm/_tqdm.py", line 878, in __exit__
    self.close()
File "/home/yury/.pyenv/versions/3.6.5/lib/python3.6/site-packages/tqdm/_tqdm.py", line 1087, in close
    self._decr_instances(self)
File "/home/yury/.pyenv/versions/3.6.5/lib/python3.6/site-packages/tqdm/_tqdm.py", line 446, in _decr_instances
    if inst.pos > abs(instance.pos):
AttributeError: 'tqdm' object has no attribute 'pos'
```

I've added `time.sleep(0.1)`. It definitely reduced the amount of errors but did not completely eliminate them.

Versions:
```
4.23.4 3.6.5 (default, Jun  4 2018, 19:06:48) 
[GCC 5.4.0 20160609] linux
```

**Comment by RLittinNV** [comment]
We have seen the same issue with concurrency.  My work around was to set "tqdm.pos = 0" before first use of tqdm.
Example code that shows it occurring regularly:

```
import time
from tqdm import *
from concurrent import futures

def testfunction(i):
    tqdm.write("started {}".format(i))
    for x in tqdm(range(100*i), desc="name"+str(i)):
        time.sleep(0.001)
    tqdm.write("finished {}".format(i))
    return i

        
wait_for = []
#tqdm.pos = 0
with futures.ThreadPoolExecutor(max_workers=10) as tpe:
    for i in range(10):
        wait_for.append(tpe.submit(testfunction, i))

for f in futures.as_completed(wait_for):
    tqdm.write("compete {}".format(f.result()))
```

**Comment by william-silversmith** [comment]
I did the same and it worked. See duplicate issues #323 another duplicate is #510. I have a similar demo here: #544 

I believe setting pos to 0 isn't a work around, but an actual solution. It's very reasonable for the default value to be zero on first use. In #544 I described the etiology of the race condition:

```
cls._instances were being generated asynchronously and tqdm._get_free_pos
was being called between __new__ and __init__. _get_free_pos was accessing
the tqdm.pos attribute which was going to be set in __init__.
```

**Comment by casperdcl** [comment]
Thanks @william-silversmith - I'll try to use your solution with unit tests. It'll likely break some positioning logic so may require other changes

**Comment by casperdcl** [comment]
happy to reopen if this is still an issue for `tqdm>4.29.0` - I think it should be fixed by #573 (78227f92dbafc97ea08e19b442d6d33f63591bf4)
